### PR TITLE
N21-408 Update external tool

### PR DESCRIPTION
**Pull request by mrikallab**
# Description
Replaced put to post for updating tools

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/N21-408
https://ticketsystem.dbildungscloud.de/browse/N21-511

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by arnegns — APPROVED** [review]

File: controllers/ctltools.js [hunk 1]
```diff
@@ -97,7 +100,7 @@ const getDeleteHandler = (req, res, next) => {
 };
 
 const getCreateHandler = (req, res, next) => {
-    req.body = sanitizeToolInputs(req.body, true);
+    req.body = sanitizeToolInputs(undefined, req.body, true);
```
Comment (CeEv) [inline controllers/ctltools.js:103]: I do not like the true / false params. But is not part of the PR, only moved. 
But if someone need it, it can be add one method with true/false as parameter and 2 others one for false one for true with speaking names that the code say what is going on.

File: controllers/ctltools.js [hunk 2]
```diff
@@ -66,9 +69,9 @@ const sanitizeToolInputs = (body, create= false) => {
 };
 
 const getUpdateHandler = (req, res, next) => {
-    req.body = sanitizeToolInputs(req.body);
+    req.body = sanitizeToolInputs(req.params.id, req.body);
 
-    api(req, { version: 'v3' }).put(`/tools/${req.params.id}`, {
+    api(req, { version: 'v3' }).post(`/tools/${req.params.id}`, {
```
Comment (CeEv) [inline controllers/ctltools.js:74]: The post it what you want for a update?

**Review by CeEv — CHANGES_REQUESTED** [review]

File: controllers/ctltools.js (hunk 2 above)
Comment (mrikallab) [reply controllers/ctltools.js:74]: yes because we want to increase version number in future, so it's not idempotent.

File: controllers/ctltools.js [hunk 3]
```diff
@@ -20,7 +20,8 @@ const trimWhitespaces = (object) => {
     });
 };
 
-const sanitizeToolInputs = (body, create= false) => {
+const sanitizeToolInputs = (id, body, create= false) => {
+    body.id = id || undefined;
```
[the remark this answers is not in the document]
Comment (arnegns) [reply controllers/ctltools.js:24]: correct, thanks 👍

File: controllers/ctltools.js (hunk 2 above)
Comment (CeEv) [reply controllers/ctltools.js:78]: ok

**Review by CeEv — APPROVED** [review]
